openadr-vtn/src/jwt.rs

```diff
@@ -19,20 +19,83 @@ pub struct JwtManager {
     decoding_key: DecodingKey,
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
-#[allow(clippy::upper_case_acronyms)]
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(tag = "role", content = "id")]
 pub enum AuthRole {
-    BL,
-    VEN,
+    UserManager,
+    Business(Option<String>),
+    VEN(String),
+}
+
+impl AuthRole {
+    pub fn is_business(&self) -> bool {
+        matches!(self, AuthRole::Business(_))
+    }
+
+    pub fn is_ven(&self) -> bool {
+        matches!(self, AuthRole::VEN(_))
+    }
+
+    pub fn is_user_manager(&self) -> bool {
+        matches!(self, AuthRole::UserManager)
+    }
 }
 
 #[derive(Debug, serde::Serialize, serde::Deserialize)]
 pub struct Claims {
     exp: usize,
     nbf: usize,
     pub sub: String,
-    pub role: AuthRole,
-    pub ven: Option<String>,
+    pub roles: Vec<AuthRole>,
+}
+
+impl Claims {
+    pub fn ven_ids(&self) -> Vec<String> {
+        self.roles
+            .iter()
+            .filter_map(|role| {
+                if let AuthRole::VEN(id) = role {
+                    Some(id.clone())
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    pub fn business_ids(&self) -> (Vec<String>, bool) {
```

Comment:
Would this be better as an `enum Foo {Any, Specific(Vec<String>)}`?

Reply:
Good idea! Will change!